adapters/repos/db/index.go

```diff
@@ -1007,15 +1007,16 @@ func (i *Index) objectVectorSearch(ctx context.Context, searchVector []float32,
 		return nil, nil, err
 	}
 
-	if len(shardNames) == 1 {
-		return out, dists, nil
-	}
-
-	if len(shardNames) > 1 && len(sort) > 0 {
-		return i.sort(out, dists, sort, limit)
+	var err error
+	if len(sort) > 0 {
+		out, dists, err = i.sort(out, dists, sort, limit)
+		if err != nil {
+			i.logger.WithField("action", "sort_vector_search_results").Error(err.Error())
+		}
+	} else if len(shardNames) > 0 {
+		out, dists = newDistancesSorter().sort(out, dists)
```

Comment:
shouldn't this kick in only only when there's a multishard setup? now it runs every time

Reply:
This change has been reverted. The nearText sort already worked without this change